src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -81,4 +122,124 @@ private bool StripAssembly(ITaskItem assemblyItem)
         return true;
     }
 
+    private bool TrimMethods(ITaskItem methodTokenFileItem)
```

Comment:
This function is doing too much.  It has at least 6 responsibilities:
1. Open the token file and read the first line
2. Determine the names for the input assembly, and the destination where it will be written
3. Open the assembly for reading
4. Read the metadata token file and the assembly method definitions and compute the hash of methods to zero out
5. Open the memory stream for writing and actually zero out the methods
6. Copy the memory stream to the destination file
7. Move files to the final place

Each of these things should be a separate function.

Reply:
How about now?